### PR TITLE
MINT-191 Set Exchange prices in ether, not wei

### PR DESCRIPTION
**Pull request by sashaaro**

### CODE REVIEW

File: src/dao/OtherContractsDAO.js [hunk 1]
```diff
@@ -118,14 +118,18 @@ class OtherContractsDAO extends AbstractMultisigContractDAO {
...
+
+      return dao.getData(TX_SET_PRICES, [buyPrice, sellPrice]).then(data => {
+        let infoArgs = {
           [TX_SET_PRICES]: '',
           contract: model.name(),
           address: model.address(),
-          buyPrice: this._c.toWei(model.buyPrice()),
-          sellPrice: this._c.toWei(model.sellPrice())
-        })
+          buyPrice: buyPrice,
```
Comment (dkchv) [inline src/dao/OtherContractsDAO.js]: use es6 shorthand syntax

File: src/components/forms/settings/other/ExchangeForm.js [hunk 2]
```diff
@@ -4,9 +4,15 @@ import { Field, reduxForm } from 'redux-form/immutable'
...
-})
+const mapStateToProps = (state) => () => {
+  const settings = state.get('settingsOtherContracts').selected.settings()
+  return { initialValues: {
+    buyPrice: settings.buyPrice.toFixed(),
+    sellPrice: settings.sellPrice.toFixed()
+  }}
+}
+
+const weiPattern = '[0-9]+([\\.][0-9]{1,18})?'
```
Comment (bshevchenko) [inline src/components/forms/settings/other/ExchangeForm.js]: weiPattern > ethPattern

File: specs/models/contracts/ExchangeContractModel.spec.js [hunk 3]
```diff
@@ -44,4 +44,11 @@ describe('exchange contract model', () => {
       'sellPrice': null
     })
   })
+
+  it('should not validate', () => {
+    const values = new Map()
+    values.set('buyPrice', 20)
+    values.set('sellPrice', 10)
+    expect(validate(values).sellPrice.length).toEqual(1)
```
Comment (bshevchenko) [inline specs/models/contracts/ExchangeContractModel.spec.js]: expect(validate(values).sellPrice.length).toEqual(1) > expect(validate(values).sellPrice).not.toEqual(null)

File: specs/models/contracts/ExchangeContractModel.spec.js [hunk 4]
```diff
@@ -44,4 +44,11 @@ describe('exchange contract model', () => {
       'sellPrice': null
     })
   })
+
+  it('should not validate', () => {
```
Comment (bshevchenko) [inline specs/models/contracts/ExchangeContractModel.spec.js]: should not validate when prices is invalid

File: src/dao/OtherContractsDAO.js [hunk 5]
```diff
@@ -118,14 +118,18 @@ class OtherContractsDAO extends AbstractMultisigContractDAO {
...
+
+      return dao.getData(TX_SET_PRICES, [buyPrice, sellPrice]).then(data => {
+        let infoArgs = {
           [TX_SET_PRICES]: '',
           contract: model.name(),
           address: model.address(),
-          buyPrice: model.buyPrice(),
-          sellPrice: model.sellPrice()
-        })
+          buyPrice: buyPrice,
```
Comment (bshevchenko) [inline src/dao/OtherContractsDAO.js]: use shorthand syntax

File: src/redux/settings/otherContracts.js [hunk 6]
```diff
@@ -136,7 +136,8 @@ export const saveContractSettings = (contract: AbstractOtherContractModel) => di
       if (!result) {
         dispatch(showContractError(contract.address()))
       }
-    }).catch(() => {
+    }).catch((e) => {
+      console.error('error', e)
```
Comment (bshevchenko) [inline src/redux/settings/otherContracts.js]: remove this debug line

File: src/utils/Web3Converter.js [hunk 7]
```diff
@@ -9,17 +10,36 @@ class Web3Converter {
...
+    if (n === null) {
+      return n
+    }
+    const isBigNumber = n.toFraction
+    if (isBigNumber) {
+      n = new BigNumber(n.toFixed())
+    }
+    let returnValue = (new BigNumber(n)).times(1000000000000000000)
+
+    return isBigNumber ? returnValue : returnValue.toString(10)
```
Comment (bshevchenko) [inline src/utils/Web3Converter.js]: DON'T REPEAT YOURSELF
diff between this function and fromWei is only in _times_ and _dividedBy_

File: src/utils/Web3Converter.js [hunk 8]
```diff
@@ -1,25 +1,49 @@
...
 import bs58 from 'bs58'
+import BigNumber from 'bignumber.js'
+
+const weiRate = 1000000000000000000
 
 class Web3Converter {
   /**
+   * @param n
+   * @param {Boolean} toWei
+   * @return {*}
```
Comment (bshevchenko) [inline src/utils/Web3Converter.js]: write @return**s**

File: src/utils/Web3Converter.js [hunk 9]
```diff
@@ -1,25 +1,49 @@
 import web3utils from 'web3/lib/utils/utils'
 import bs58 from 'bs58'
+import BigNumber from 'bignumber.js'
+
+const weiRate = 1000000000000000000
 
 class Web3Converter {
   /**
+   * @param n
+   * @param {Boolean} toWei
```
Comment (bshevchenko) [inline src/utils/Web3Converter.js]: don't specify types in your jsdoc
with flow you can do this right inside of your function signatures

File: src/utils/Web3Converter.js [hunk 10]
```diff
@@ -1,25 +1,49 @@
...
+   * @return {*}
+   * @private
+   */
+  _convert (n: number | string | BigNumber, toWei = true) {
```
Comment (bshevchenko) [inline src/utils/Web3Converter.js]: name it _weiConvert

**Pull request merged**